Assets/Scripts/ScriptableObjects/Inventory.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "NewInventory", menuName = "Scriptable Objects/Inventory")]
+public class Inventory : ScriptableObject
+{
+    [Range(0,999)]
+    public int coins;
+    [Range(0,9)]
+    public int potions;
+
+    //Limit the number of coins
+    public void CheckCoins()
+    {
+        coins = Mathf.Clamp(coins, 0, 999);
```

Comment:
Is this required even after setting the Range attribute?

Reply:
Range attribute doesn't allow you  to set a number out of range with unity inspector, while a code sequence can.